dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tracker/MetadataItemsHandler.java

```diff
@@ -85,6 +90,10 @@ public class MetadataItemsHandler {
 
   protected final UserService userService;
 
+  protected final OrganisationUnitService organisationUnitService;
+
+  protected final OrganisationUnitResolver organisationUnitResolver;
```

Comment:
I think we could add the OrganisationUnitService into the OrganisationUnitResolver, and encapsulate this code into a method? So, we only need to inject the "resolver" in MetadataItemsHandler. In this way, only the OrganisationUnitResolver will have access to OrganisationUnitService.
```
organisationUnitService.getOrganisationUnitsByUid(orgUnitIds).stream()
        .collect(
            Collectors.toMap(OrganisationUnit::getUid, orgUnit -> toMetadataItem(orgUnit, params)));
```